File: api/src/endpoint/repository/get_repository_file.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use super::default_ref;
use crate::{endpoint::Endpoint, resource::repository::RepositoryFileResource};

pub struct GetRepositoryFile;

impl Endpoint for GetRepositoryFile {
    const PATH: &'static str = "/repository/{owner}/{repo}/file";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryFileRequest;
    type Response = GetRepositoryFileResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryFileRequest {
    #[serde(default = "default_ref")]
    pub ref_name: String,
    pub path: String,
}

pub type GetRepositoryFileResponse = RepositoryFileResource;

File: gitdot-api-ts/src/endpoint/repository/get-repository-file.ts
import { z } from "zod";
import { RepositoryFileResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetRepositoryFileRequest = z.object({
  ref_name: z.string(),
  path: z.string(),
});
export type GetRepositoryFileRequest = z.infer<typeof GetRepositoryFileRequest>;

export const GetRepositoryFileResponse = RepositoryFileResource;
export type GetRepositoryFileResponse = z.infer<
  typeof GetRepositoryFileResponse
>;

export const GetRepositoryFile = {
  path: "/repository/{owner}/{repo}/file",
  method: "GET",
  request: GetRepositoryFileRequest,
  response: GetRepositoryFileResponse,
} as const satisfies Endpoint;
export type GetRepositoryFile = typeof GetRepositoryFile;
